coreax/coresubset.py

```diff
@@ -151,6 +153,9 @@ def fit_to_size(self, coreset_size: int) -> None:
 
         :param coreset_size: The size of the of coreset to generate
         """
+        if coreset_size == 0:
+            raise ValueError("coreset_size must be non-zero")
+
```

Comment:
Can this be removed if we make the suggested changes to line 187 (see later comment)

Reply:
Done for herding. However, Stein thinning does need a check-for-nonpositive-integer if regularisation is used (see below).